src/promptflow/tests/sdk_cli_test/recording_utilities/mocked_functions.py

```diff
@@ -0,0 +1,166 @@
+            node.connection == "azure_open_ai_connection" or node.provider == "AzureOpenAI"
+        ):
+            prompt_tpl = self._load_source_content(node)
+            prompt_tpl_inputs = get_inputs_for_prompt_template(prompt_tpl)
+            callable = partial(just_return, "AzureOpenAI", prompt_tpl, prompt_tpl_inputs, recording_file)
+            return ResolvedTool(node=node, definition=None, callable=callable, init_args={})
+        else:
+            return None
+
+    def _mock_toolresolver_resolve_tool_by_node(self, node: Node, convert_input_types=True) -> ResolvedTool:
+        try:
+            if node.source is None:
+                raise UserErrorException(f"Node {node.name} does not have source defined.")
+
+            if node.type is ToolType.PYTHON:
+                if node.source.type == ToolSourceType.Package:
+                    return self._resolve_package_node(node, convert_input_types=convert_input_types)
+                elif node.source.type == ToolSourceType.Code:
+                    return self._resolve_script_node(node, convert_input_types=convert_input_types)
+                raise NotImplementedError(f"Tool source type {node.source.type} for python tool is not supported yet.")
+            elif node.type is ToolType.PROMPT:
+                return self._resolve_prompt_node(node)
+            elif node.type is ToolType.LLM:
+                resolved_tool = _resolve_replay_node(self, node, convert_input_types=convert_input_types)
+                if resolved_tool is None:
+                    resolved_tool = self._resolve_llm_node(node, convert_input_types=convert_input_types)
+                return resolved_tool
+            elif node.type is ToolType.CUSTOM_LLM:
+                if node.source.type == ToolSourceType.PackageWithPrompt:
+                    resolved_tool = self._resolve_package_node(node, convert_input_types=convert_input_types)
+                    return self._integrate_prompt_in_package_node(node, resolved_tool)
+                raise NotImplementedError(
+                    f"Tool source type {node.source.type} for custom_llm tool is not supported yet."
+                )
+            else:
+                raise NotImplementedError(f"Tool type {node.type} is not supported yet.")
+        except Exception as e:
+            if isinstance(e, PromptflowException) and e.target != ErrorTarget.UNKNOWN:
+                raise ResolveToolError(node_name=node.name, target=e.target, module=e.module) from e
+            raise ResolveToolError(node_name=node.name) from e
+
+    return _mock_toolresolver_resolve_tool_by_node
+
+
+def mock_get_local_connections_from_executable(executable: Flow, client):
+    # Mock of get_connections, it should not return any connection in replay mode.
+    connection_names = executable.get_connection_names()
+    result = {}
+    for n in connection_names:
+        try:
+            conn = client.connections.get(name=n, with_secrets=True)
```

Comment:
why do we need to mock this?

Reply:
Good Point. After replaying, this is useless.